File: Benchmarks/SpannerComparison.cs
using System;
using System.Text.Patterns;
using System.Text.RegularExpressions;
using BenchmarkDotNet.Attributes;

namespace Benchmarks {
	[ClrJob, CoreJob, CoreRtJob]
	[MemoryDiagnoser]
	public class SpannerComparison {

		readonly Pattern Spanner = +(Pattern)"Hi!";

		readonly Regex Regex = new Regex("^(Hi!)+");

		[Params("Hi!Hi!", "Hi!Hi!Hi!", "Hi!", "Okay?")]
		public String Source { get; set; }

		[Benchmark(Baseline = true)]
		public Result Stringier() => Spanner.Consume(Source);

		[Benchmark]
		public Match MSRegex() => Regex.Match(Source);

	}
}

File: Benchmarks.FSharp/SpannerComparison.fs
namespace Benchmarks

open System.Text.Patterns
open FParsec
open BenchmarkDotNet.Attributes
open BenchmarkDotNet.Diagnosers

[<ClrJob>]
[<MemoryDiagnoser>]
type SpannerComparison() =
    let pattern = +"Hi!"
    let parsec = attempt (many1Strings (pstringCI "Hi!"))

    [<Params("Hi!Hi!", "Hi!Hi!Hi!", "Hi!", "Okay?")>]
    member val Source = "" with get, set

    [<Benchmark>]
    member this.PatternConsume() = pattern.Consume(this.Source)

    [<Benchmark(Baseline = true)>]
    member this.ParsecRun() = run parsec this.Source